pkg/config/log.go

```diff
@@ -101,55 +108,23 @@ func SetupLogger(loggerName LoggerName, logLevel, logFile, syslogURI string, sys
 		}
 	}
 
-	formatID := "common"
-	if jsonFormat {
-		formatID = "json"
-	}
-
-	configTemplate := `<seelog>`
-	configTemplate += fmt.Sprintf(`<outputs formatid="%s">`, formatID)
-
-	if logToConsole {
-		configTemplate += `<console />`
-	}
-	if logFile != "" {
-		configTemplate += fmt.Sprintf(`<rollingfile type="size" filename="%s" maxsize="%d" maxrolls="%d" />`, logFile, Datadog.GetSizeInBytes("log_file_max_size"), Datadog.GetInt("log_file_max_rolls"))
-	}
+	seelogConfig = NewSeelogConfig(string(loggerName), seelogLogLevel, formatID, BuildJSONFormat(loggerName), BuildCommonFormat(loggerName), syslogRFC)
```

Comment:
The `seelogConfig` init could probably be done before the `if syslogURI != ""` block; this way, you will be able to discard the `syslog` bool and directly call the `seelogConfig.configureSyslog(syslogURI, useTLS)` in the syslogURI block.

Reply:
Will account for both comment.